src/energyplus/ForwardTranslator/ForwardTranslateCoilSystemCoolingWater.cpp

```diff
@@ -74,36 +74,32 @@ namespace energyplus {
           idf->setString(Coil_Cooling_WaterFields::AirInletNodeName, airInletNodeName);
           idf->setString(Coil_Cooling_WaterFields::AirOutletNodeName, airOutletNodeName);
           // Add IddObjectType::Coil_Cooling_Water_DetailedGeometry if implemented
+        } else if (idf->iddObject().type() == IddObjectType::CoilSystem_Cooling_Water_HeatExchangerAssisted) {
+          // The logic is built in the translateCoilSystemCoolingWaterHeatExchangerAssisted method
+          // Essentially it's the same as getting the HX (CoilSystem_Cooling_Water_HeatExchangerAssistedFields::HeatExchangerName)
+          // and setting HeatExchanger_AirToAir_SensibleAndLatentFields::SupplyAirInletNodeName & ExhaustAirOutletNodeName
         } else {
-          // Shouldn't happen, accepts only Coil:Cooling:Water or Coil:Cooling:Water:DetailedGeometry
-          // Shouldn't happen, accepts only Coil:Cooling:DX:SingleSpeed or Coil:Cooling:DX:VariableSpeed
-          LOG(Fatal, modelObject.briefDescription()
-                       << " appears to have a cooling coil that shouldn't have been accepted: " << coolingCoil.briefDescription());
-          OS_ASSERT(false);
-        }
-      }
-      if (auto coilCoolingWater = coolingCoil.optionalCast<CoilCoolingWater>()) {
-        if (auto controller = coilCoolingWater->controllerWaterCoil()) {
-          if (auto idf = translateAndMapModelObject(controller.get())) {
-            idf->setString(Controller_WaterCoilFields::SensorNodeName, airOutletNodeName);
-          }
+          // Shouldn't happen, accepts only Coil:Cooling:Water CoilSystem:Cooling:Water:HeatExchangerAssisted (or Coil:Cooling:Water:DetailedGeometry, not wrapped)
+          LOG_AND_THROW(modelObject.briefDescription()
+                        << " appears to have a cooling coil that shouldn't have been accepted: " << coolingCoil.briefDescription());
         }
       }
 
-      // Need a SPM:MixedAir on the Coil:Cooling:Water outlet node (that we **created** just above in IDF directly, so it won't get picked up by the
+      // TODO: is that true?!
+      // Need a SPM:MixedAir on the outlet node (that we **created** just above in IDF directly, so it won't get picked up by the
```

Comment:
@joseph-robertson not sure if that's just copy pasted fromCoilSystem_Cooling_Water_HeatExchangerAssisted or actually required.